command/agent/http.go

```diff
@@ -595,7 +594,7 @@ func setMeta(resp http.ResponseWriter, m *structs.QueryMeta) {
 // setHeaders is used to set canonical response header fields
 func setHeaders(resp http.ResponseWriter, headers map[string]string) {
 	for field, value := range headers {
-		resp.Header().Set(http.CanonicalHeaderKey(field), value)
```

Comment:
This is a function, not a cast... I don't think we want to remove this.  See https://golang.org/pkg/net/http/#CanonicalHeaderKey

Reply:
This was reported as redundant, as Header.Set already canonicalizes it: https://staticcheck.io/docs/checks#S1035 and https://golang.org/src/net/http/header.go#L31

Reply:
Oh sweet.